app/src/main/java/movierating/repositories/MovieRepository.java

```diff
@@ -0,0 +1,18 @@
+package movierating.repositories;
+
+import movierating.models.Movie;
+
+import java.util.List;
+
+public class MovieRepository implements  Repository{
```

Comment:
It is good that we are having a common super structure as Repository. But is it necessary to have common hierarchy. It is not necessary that they have some common methods to be used right?

Reply:
If you are able to bring out some common hierarchy and structure to all the repositories. Then it totally makes sense to implement it. But right now, since it is not there I dont think it is necessary.

Reply:
Removed the Repository Interface now.